crates/ra_hir_ty/src/tests/traits.rs

```diff
@@ -1989,9 +1983,75 @@ fn test<I: Iterator<Item: Iterator<Item = u32>>>() {
 }
 "#,
     );
-    // assert_eq!(t, "u32");
-    // doesn't currently work, Chalk #234
-    assert_eq!(t, "{unknown}");
+    assert_eq!(t, "u32");
+}
+
+#[test]
+fn proc_macro_server_types() {
```

Comment:
ah btw this is code from RA that isn't working yet; even though we seem to be able to expand the macro, if I write the traits without macros I can't reproduce the problem. So it might be some macro issue, I'm not sure. cc @edwin0cheng

Reply:
I just checked, it is due to we still not support mbe in trait items yet :) I would open #4039 to track the issue.